### ISSUE TITLE
After selecting a file, mouse wheel does not scroll until title bar is clicked (Windows 11)

### ISSUE THREAD

**Issue by DaleSchultz** [TheGameCreators/AGKIDE#14]
I have a project with a number of files.
If I click on one of the files in the project tree and then click in the code window then roll the mouse wheel, it does not scroll.
If I then click in the title bar of AGK then roll the mouse wheel it scrolls as it should, until another file is selected.
It is as if AGK does not know it has focus and does not do anything with the mouse events.

AGK IDE 2021.10.19

Windows machine:

Processor	Intel(R) Core(TM) i3-10105 CPU @ 3.70GHz   3.70 GHz
Installed RAM	8.00 GB (7.74 GB usable)
Device ID	1AFBDC5D-383C-46AD-9423-4E599AEE33E2
Product ID	00342-21930-62404-AAOEM
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

**Comment by DaleSchultz** [comment]
In addition, the toolbar icons _also_ ignore the mouse clicks until the title bar has been clicked again.

Mouse clicks on the tabs "Project/Symbols/Debug" as well as the message window all work fine _even_ when in the 'bad' state.

Sometimes even multiple clicks in the title bar and menu bar don't resolve the problem.
Pressing F5 runs the program (or ends it) so the application has focus.

This makes it extremally hard to work with.

Until yesterday I was working with Windows 10 and not having these issues.
New computer, fresh install of AGK.

**Comment by DaleSchultz** [comment]
another clue...
After clicking a file in the project tree, the problem starts. If one simply moves the mouse cursor _outside the AGK window_, and then back over it again, it comes right again. So clicking a file, causes it to lose 'ownership' of the mouse events.

This is why clicking the title bar seemed to fix it, the mouse has gone 'beyond' AGK and on return it handles the mouse events.

**Comment by DaleSchultz** [comment]
update:

I rebooted and then started only AGK. Worked perfectly.
I then slowly added other running apps until the problem came back. When a VB6 app that hooks the mouse wheel events is running, the issue returns. So somehow the VB6 app and AGK (but not other apps) were in conflict.

I can no longer reproduce this at will. Even if I run the VB6 program! Every time I think I have nailed down the scenario to reproduce it, some time later, it all changes again.

So I suggest wait and see if others report similar issues and if I can reproduce this again perhaps with a demo program, I'll revisit this issue.

**Comment by DaleSchultz** [comment]
Some more progress in narrowing down where the conflict is coming from.

Inside Visual Basic 6 IDE there is a standard Add-in called Visual Components Manager
If that Add-in is enabled/loaded then AGK loses mouse events (scroll, mouse over,  and click at least) after one clicks on a file in a project tree.

The VB6 IDE just has to be running, with an empty project, not actually running any application. The two apps are even running on separate screens so no screen overlap even!

The bug(s) may lie in whatever that Add-in does inside VB6, or the way Windows 11 interacts with it, or AGK.
So for now, the workaround seems to be to disable/unload that Add-in in VB6

**Comment by DaleSchultz** [comment]
well you probably all think I am 100% bonkers. Before I post anything here **I test and check that what I am seeing is reproducible.**

Now, not even disabling the Visual components manager fixes it. Now just having VB6 IDE running causes the issue, with no workaround.

It is something very complex, order of loading, window handle numbers, who knows?

**Comment by adambiser** [comment]
I'm on Windows 10 and don't have this problem in this issue.

[The VB6 IDE is notoriously bad at handling the mouse wheel.](https://docs.microsoft.com/en-us/troubleshoot/dotnet/visual-basic/mouse-wheel-events-do-not-work)  I use VB6IDEMouseWheelAddin mentioned on that page.
As a side note, I also use the MZ Tools add-in and it sometimes captures hotkeys when the IDE doesn't have focus.

What are you using to get the mousewheel to work in the VB6 IDE?  Aside from the two add-ins I've mentioned, I only have those that are installed by the VB6 IDE.  I do not see "Visual Components Manager" in my list.

**Comment by DaleSchultz** [comment]
yes, I also use that mouseWheel add-in and initially assumed that was the cause but I disabled that too!
It defies clean reproduction, after disabling all the add-ins, I got the behavior again.
I even check my registry setting against that page too, to no avail.
It is something else....
I was using Windows 10 until this week, with VB6 and AGK and did not have these problems, so it seems related to windows 11

**Comment by adambiser** [comment]
I won't be updating to Windows 11 for some time, so I can't be of more help.

I know when MZ Tools is catch hotkeys when the IDE doesn't have focus, I switch back to VB6, click the open code window, then go back to the other program.  Might not be helpful here.

**Comment by DaleSchultz** [comment]
thanks I'll try that when it happens again!

**Comment by DaleSchultz** [comment]
`I do not see "Visual Components Manager`

Sorry missed the question earlier... I checked it when running the install for VB6 so it something native to VB6 and I have no clue what it does. Anyway, disabled now.
